rules/integrations/azure/credential_access_azure_entra_totp_brute_force_attempts.toml

```diff
@@ -0,0 +1,158 @@
+
+#### Response and Remediation:
+
+- **Immediate Actions**:
+    - If proven malicious, lock the affected account temporarily to prevent further unauthorized attempts.
+    - Notify the user of suspicious activity and validate their access to the account.
+    - Reset passwords and MFA settings for the affected user to prevent unauthorized access while communicating with the user.
+
+- **Strengthen Authentication Policies**:
+    - Ensure conditional access policies are configured to monitor and restrict anomalous login behavior.
+    - Consider a different MFA method or additional security controls to prevent future bypass attempts.
+
+- **Monitor and Audit**:
+    - Implement additional monitoring to track high-frequency authentication failures across the environment.
+    - Audit historical logs for similar patterns involving other accounts to identify broader threats.
+
+- **Educate and Train Users**:
+    - Provide guidance on the secure use of MFA and the importance of recognizing and reporting suspicious activity.
+"""
+references = [
+    "https://www.oasis.security/resources/blog/oasis-security-research-team-discovers-microsoft-azure-mfa-bypass",
+    "https://learn.microsoft.com/en-us/entra/identity/",
+    "https://learn.microsoft.com/en-us/entra/identity/monitoring-health/concept-sign-ins",
+]
+risk_score = 47
+rule_id = "3fac01b2-b811-11ef-b25b-f661ea17fbce"
+setup = """#### Required Azure Entra Sign-In Logs
+This rule requires the Azure logs integration be enabled and configured to collect all logs, including sign-in logs from Entra. In Entra, sign-in logs must be enabled and streaming to the Event Hub used for the Azure logs integration.
+"""
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Domain: SaaS",
+    "Data Source: Azure",
+    "Data Source: Entra ID",
+    "Data Source: Entra ID Sign-in",
+    "Use Case: Identity and Access Audit",
+    "Use Case: Threat Detection",
+    "Tactic: Credential Access",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-azure.signinlogs* metadata _id, _version, _index
+| where
+    // filter for Entra Sign-In Logs
+    event.dataset == "azure.signinlogs"
+    and azure.signinlogs.operation_name == "Sign-in activity"
+
+    // filter for MFA attempts with OATH conditional access attempts or TOTP
+    and azure.signinlogs.properties.authentication_requirement == "multiFactorAuthentication"
+    and azure.signinlogs.properties.mfa_detail.auth_method == "OATH verification code"
+
+    // filter on failures only from brute-force attempts
+    and azure.signinlogs.properties.conditional_access_status == "failure"
+    and azure.signinlogs.result_description == "Authentication failed during strong authentication request."
+| keep azure.signinlogs.properties.sign_in_identifier
+| stats
+    // aggregate by the sign-in account or principal
+    failed_totp_code_attempts = count(*) by azure.signinlogs.properties.sign_in_identifier
```

Comment:
Probably doesn't matter for this rule but would be neat if we could pull in IP context.

Reply:
Great thought! The reason I didn't aggregate on source too is because of the popularity in residential proxies. It may not hit the threshold if the same actor is using various residential proxies while brute forcing.